azure.go

```diff
@@ -447,3 +448,12 @@ func graphURIFromName(name string) (string, error) {
 
 	return c, nil
 }
+
+var nameRx = regexp.MustCompile(`^([a-zA-Z][a-zA-Z0-9\-]*)$|^$`)
```

Comment:
From what I can tell, a typical subscription ID looks like `1234abcd-1234-1234-abcd-12345abcdefg` which this regex will fail to match https://go.dev/play/p/lVWlfZNc42w

Reply:
You're right, i was just being sloppy, it's "pretty obviously" a guid, although documentation is a bit thin on the ground stating it directly.

I was able to find this: https://learn.microsoft.com/en-us/rest/api/defenderforcloud/tasks/get-subscription-level-task